File: 119/getRow.php
<?php
class Solution {

    /**
     * @param Integer $rowIndex
     * @return Integer[]
     */
    function getRow($rowIndex) {
        $data = [];
        $data[1] = [1];
        for($i=2;$i<=$rowIndex+1;$i++){
            $preRow = $data[$i-1];
            $temp = [];
            $temp[0] = $preRow[0];
            for ($j=1;$j<$i-1;$j++){
                $temp[$j] = $preRow[$j-1]+$preRow[$j];
            }
            $temp[$i-1] = $preRow[count($preRow)-1];
            $data[$i] = $temp;
        }
        return $data[$rowIndex+1];
    }
}
$obj = new Solution();
$rowIndex = 3;
$result = $obj->getRow($rowIndex);
print_r($result);

File: src/Array/119_杨辉三角II/getRow.go
package _19_杨辉三角II

func getRow(rowIndex int) []int {
	if rowIndex+1 == 1 {
		return []int{1}
	}
	if rowIndex+1 == 2 {
		return []int{1, 1}
	}
	preRow := []int{1, 1}
	for i := 3; i <= rowIndex+1; i++ {
		temp := make([]int, i)
		temp[0] = 1
		temp[i-1] = 1
		for j := 1; j < i-1; j++ {
			temp[j] = preRow[j-1] + preRow[j]
		}
		preRow = temp
	}

	return preRow
}
